src/OpenSage.Game/Graphics/ParticleSystems/ParticleSystem.cs

```diff
@@ -432,6 +403,26 @@ private void UpdateParticle(ref Particle particle)
             particle.Timer += 1;
         }
 
+        private static void FindKeyframes<T>(int timer,
+            IReadOnlyList<T> keyFrames,
+            out T next, out T prev)
```

Comment:
This could return a tuple instead of having two out params, but this resulted in cleaner code. Performance should be identical.